api/src/main/java/marquez/db/FlywayFactory.java

```diff
@@ -74,6 +76,15 @@ public final class FlywayFactory {
   @Getter @Setter
   private String repeatableSqlMigrationPrefix = DEFAULT_REPEATABLE_SQL_MIGRATION_PREFIX;
 
+  @Getter private Optional<String> schema = Optional.empty();
```

Comment:
In Marquez, we don't use `Optional`s for replacements of `null`s, also the `Optional` class is not serializable (see [How to use Optionals In Java](https://dzone.com/articles/optional-in-java)). What's common throughout the codebase is to define a getter method as follows:

```java
public Optional< String > getSchema() {
  return Optional.ofNullable(schema);
}
```

Reply:
I think those guidelines are sensible when dealing with service models that are serialized and contain properties that may be accessed by other parts of the code. This field, however, is entirely internal and never accessed by any other object. We don't access any other fields via public getter method (e.g., there's no `getRepeatableSqlMigrationPrefix()` in the previous line). So it doesn't seem consistent or useful to use a getter here.

I could simply default the field to `null` and replace the `Optional` call with `schema == null ? new String[]{} : new String[]{schema}`, but I think `Optional` is preferable here. 

I could use `Optional<String> schemaOption = Optional.ofNullable(schema)`, but that's really just doing `schema == null ? Optional.empty() : Optional.of(schema)`, which is not really different from the previous option. 

Of the options, I think using the `Optional.empty()` field as a default is the cleanest. It's not really different from using `Optional` as a replacement for `null` inside a method body, which we do in Marquez all the time. There's no serialization concern, no external client that needs to access the state of this object...

Reply:
That's a fair distinction.